jetbrains-core/src/software/aws/toolkits/jetbrains/core/notification/NoticeManager.kt

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.notification
+
+import com.intellij.notification.NotificationDisplayType
+import com.intellij.notification.NotificationGroup
+import com.intellij.notification.NotificationType
+import com.intellij.notification.Notifications
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.resources.message
+
+interface NoticeManager {
+    fun getRequiredNotices(notices: List<NoticeType>, project: Project): List<NoticeType>
+    fun notify(notices: List<NoticeType>, project: Project)
+
+    companion object {
+        fun getInstance(): NoticeManager = ServiceManager.getService(NoticeManager::class.java)
+    }
+}
+
+internal const val NOTICE_NOTIFICATION_GROUP_ID = "AWS Toolkit Notices"
+
+@State(name = "notices", storages = [Storage("aws.xml")])
+class DefaultNoticeManager : PersistentStateComponent<NoticeStateList>,
+    NoticeManager {
+    private val internalState = mutableMapOf<String, NoticeState>()
+    private val notificationGroup = NotificationGroup(NOTICE_NOTIFICATION_GROUP_ID, NotificationDisplayType.STICKY_BALLOON, true)
+
+    override fun getState(): NoticeStateList = NoticeStateList(internalState.values.map { it }.toList())
+
+    override fun loadState(state: NoticeStateList) {
+        internalState.clear()
+        state.value.forEach {
+            val id = it.id ?: return@forEach
+            internalState[id] = it
+        }
+    }
+
+    /**
+     * Returns the notices that require notification
+     */
+    override fun getRequiredNotices(notices: List<NoticeType>, project: Project): List<NoticeType> = notices.filter { it.isNotificationRequired() }
+        .filter {
+            internalState[it.id]?.let { state ->
+                state.noticeSuppressedValue?.let { previouslySuppressedValue ->
+                    return@filter !it.isNotificationSuppressed(previouslySuppressedValue)
+                }
+            }
+
+            true
+        }
+
+    override fun notify(notices: List<NoticeType>, project: Project) {
+        notices.forEach { notify(it, project) }
+    }
+
+    private fun notify(notice: NoticeType, project: Project) {
```

Comment:
also take a look at `NotificationUtils`, which already has some of this logic

Reply:
I decided to leave this as is (for now) so that I have the notification object for use with the `notification.hideBalloon()` call